### ISSUE TITLE
Compatibility issue with AngularJS 1.4.1 - 1.4.3

### ISSUE THREAD

**Issue by sguidi** [MoonStorm/trNgGrid#198]
My project use last stable angular js version 1.4.3.
I tried on the online plunker example http://plnkr.co/edit/JCLrJD?p=preview to change the import AngualrJS from 1.3.1 to 1.4.3, this result in various error on the log console.

Is AngularJS 1.4.x supported by trNgGrid?
Is there any specific configuration that i'm missing?

**Comment by MoonStorm** [comment]
Thanks for reporting.
Never tried 1.4.

**Comment by franciskim** [comment]
Getting quite a few `Language identifier is not set` on 1.4.x

**Comment by irvituz1** [comment]
The same issues, can't find localization even if it is present in angular.js and can be printed by Current locale ID: us-en.
Will be glad to figure out how to fix this!

**Comment by MoonStorm** [comment]
It looks like some default TrNgGrid config values are being ignored, causing failures later on.

A similar thing happened when Angular 1.3 came out. So much for backwards compatibility when developing with Angular.

**Comment by toddhd** [comment]
Just as an FYI, I too am getting this error, and tons of them. Angular v1.4.3. What's worse is that they multiply. So for example, when I load the page with my grid, I get 40+ language errors, but every time I do "something", such as sort a column, that number increases exponentially. See this video for a better example of what's happening: http://screencast.com/t/58nnpRmlpVP
(Yes, I know you're working on it. Just letting you know. It makes it hard to find errors in my own code because they get swallowed up in thousands of language errors. )

**Comment by carueda** [comment]
I'm upgrading trNgGrid after some time and I'm just noticing this error, too.

Since I want to use the latest and greatest (in particular to test performance improvements, in part related with what is commented in #162), I will perhaps just adjust line 1258 (in my local [trNgGrid.js version](https://github.com/MoonStorm/trNgGrid/blob/c4c258f141207a0b66dd14d254ae6b97b92a1f21/src/js/trNgGrid.js#L1258)) to assign a value (that's just good enough for my purposes) instead of throwing the exception:
```js
if (!languageId) {
    languageId = 'us-en'; //throw "Language identifier is not set";
}
```

**Comment by MoonStorm** [comment]
@carueda Server-side is also busted.

I managed to narrow down the cause of the problem. Angular version 1.4.1 introduced either a bug or an undocumented breaking change when it comes to how bindings are handled. I logged a bug on their end. Depending on their answer, we'll find a way to deal with this issue. 

What's disturbing is that this was introduced in a minor version update. For the time being, if you really miss features from 1.4, my only suggestion for now is to downgrade Angular to 1.4.0. Just bear in mind there might be other compatibility issues lurking which we haven't discovered yet.

**Comment by MoonStorm** [comment]
Just got word from the Angular team that the issue has been fixed in their unreleased 1.4.4. If you want to try it out, their latest version sits here: https://code.angularjs.org/snapshot/angular.js

Until they get this version out, you should stick to 1.3.17 or 1.4.0. Just remember few people have used  the grid in 1.4. and there may still be other compatibility issues.

**Comment by carueda** [comment]
Thanks!

**Comment by toddhd** [comment]
Thank you guys - this seems to also have fixed the column display-name issue from #199. Awesome!
